java/arcs/android/sdk/host/AndroidHost.kt

```diff
@@ -29,15 +27,14 @@ import arcs.sdk.android.storage.ServiceStoreFactory
 abstract class AndroidHost(
```

Comment:
BTW, I favor keeping this abstract and requiring people to still choose AndroidHost, JvmHost, WebHost, etc. It allows to add more platform specific functionality into later, for example, Android, without needing to refactor all the API usages.

Reply:
Why can't future platform-specific functionality be provided via composition, as this PR is doing with EntityHandleManager?

Reply:
It could, anything done with inheritance can be done with composition, but I feel it's less ergonomic to make the developer have to provide platform dependencies everytime. I suppose this could be fixed by some platform independent Config objects, like

```
sealed class Config {
   object JvmConfig: Config ... HandleManager, JvmTime, Resurrector, etc
   class AndroidConfig: Config ... HandleManager, JvmTime, Resurrector, etc
   object JsConfig: Config ... HandleManager, JvmTime, Resurrector, etc
   object WasmConfig: Config ... HandleManager, JvmTime, Resurrector, etc
}
```  

And then you write

```
val arcHost = ArcHost.create(AndroidConfig.for(service, lifecycle))
```

I guess my point is, we want the API to be such that making refactors in GH don't incur breakages in G3 as often as they do now.

Reply:
Yep, I was imagining using something along the lines of your proposal there. The variant I had in mind was just to declare a few top level functions for the common usage configurations. Same overall idea, though.

Reply:
The `*Config` parameter approach you suggest makes it easier for someone using Arcs to write a subclassed ArcHost (for example, to allow access to an ingress particle), so I think I like that approach better.

Reply:
I agree. I have no objection to the refactoring.